crates/oxc_hir/src/hir_util.rs

```diff
@@ -372,3 +376,66 @@ pub fn get_boolean_value(expr: &Expression) -> Option<bool> {
         _ => None,
     }
 }
+
+/// Port from [closure-compiler](https://github.com/google/closure-compiler/blob/e13f5cd0a5d3d35f2db1e6c03fdf67ef02946009/src/com/google/javascript/jscomp/NodeUtil.java#L234)
+/// Gets the value of a node as a String, or `None` if it cannot be converted. When it returns a
+/// String, this method effectively emulates the `String()` JavaScript cast function.
+/// This method does not consider whether `expr` may have side effects.
+fn get_string_value(expr: &Expression) -> Option<String> {
```

Comment:
It is better to return &str instead of an allocated String

Reply:
It seems you need to return allocated string in some cases? Try `get_string_value<'a>(expr: &'a Expression) -> Option<Cow<'a, str>>`